src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -791,28 +370,68 @@ async function getFlows(
       ["realizedLpFeePct"]
     )
   ).filter((fill: FillWithBlock) => {
+    const _tokenSymbol = hubPoolClient.getL1TokenInfoForL2Token(fill.destinationToken, fill.destinationChainId)?.symbol;
     // We only want to include full fills as flows. Partial fills need to request refunds and those refunds
     // will be included as flows.
-    return fill.fillAmount.eq(fill.totalFilledAmount);
+    return _tokenSymbol === tokenSymbol && fill.fillAmount.eq(fill.totalFilledAmount);
   }) as UbaFlow[];
 
-  const refundRequests: UbaFlow[] = await getValidRefundCandidates(
-    chainId,
-    hubPoolClient,
-    spokePoolClients,
-    {
-      fromBlock,
-      toBlock,
-    },
-    ["realizedLpFeePct"]
-  );
+  const refundRequests: UbaFlow[] = (
+    await getValidRefundCandidates(
+      chainId,
+      hubPoolClient,
+      spokePoolClients,
+      {
+        fromBlock,
+        toBlock,
+      },
+      ["realizedLpFeePct"]
+    )
+  ).filter((refundRequest: RefundRequestWithBlock) => {
+    const _tokenSymbol = hubPoolClient.getL1TokenInfoForL2Token(
+      refundRequest.refundToken,
+      refundRequest.repaymentChainId
+    )?.symbol;
+    return _tokenSymbol === tokenSymbol;
+  });
 
   // This is probably more expensive than we'd like... @todo: optimise.
   const flows = sortEventsAscending(deposits.concat(fills).concat(refundRequests));
 
   return flows;
 }
 
+export function getBundleKeyForBlockRanges(blockRanges: number[][]): string {
+  return JSON.stringify(blockRanges.map((blockRange) => blockRange[0]));
+}
+/**
+ * The return value should be a number whose sign indicates the relative order of the two elements:
+ * negative if a is less than b, positive if a is greater than b, and zero if they are equal.
+ * @param fx
+ * @param fy
+ * @returns
+ */
+export function flowComparisonFunction(a: UbaFlow, b: UbaFlow): number {
+  if (a.blockTimestamp !== b.blockTimestamp) {
+    return a.blockTimestamp - b.blockTimestamp;
+  }
+
+  // If fx and fx have same blockTimestamp and same quote block then... FML... what do?
```

Comment:
This should only be used on flows where the tokens are the same. I like the idea of picking inflows first.

As far as tie breaking inflow vs inflow and outflow vs outflow... it might just be some arbitrary rule as you discussed above